presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/joins/JoinGraph.java

```diff
@@ -53,8 +54,25 @@
 
     public static List<JoinGraph> buildFrom(PlanNode plan)
     {
+        return buildFrom(plan, Lookup.noLookup());
+    }
+
+    /**
+     * Build a non-trivial (more than 1 element) {@link JoinGraph} containing {@code plan}.
+     */
+    public static Optional<JoinGraph> buildShallowFrom(PlanNode plan, Lookup lookup)
```

Comment:
That it builds only top-level `JoinGraph`. Compare that to the already existing `buildFrom` which builds all `JoinGraphs` from the plan tree.